cli/command/container/attach.go

```diff
@@ -64,36 +65,36 @@ func NewAttachCommand(dockerCli command.Cli) *cobra.Command {
 	}
 
 	flags := cmd.Flags()
-	flags.BoolVar(&opts.noStdin, "no-stdin", false, "Do not attach STDIN")
-	flags.BoolVar(&opts.proxy, "sig-proxy", true, "Proxy all received signals to the process")
-	flags.StringVar(&opts.detachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
+	flags.BoolVar(&opts.NoStdin, "no-stdin", false, "Do not attach STDIN")
+	flags.BoolVar(&opts.Proxy, "sig-proxy", true, "Proxy all received signals to the process")
+	flags.StringVar(&opts.DetachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
 	return cmd
 }
 
-func runAttach(dockerCli command.Cli, opts *attachOptions) error {
-	ctx := context.Background()
+// RunAttach executes an `attach` command
+func RunAttach(ctx context.Context, dockerCli command.Cli, opts *AttachOptions) error {
```

Comment:
I was exploring some options in https://github.com/docker/cli/pull/4643, and all those are probably out of scope for this PR, but perhaps if would make sense to include one part of that PR, and to pass `containerID` as separate argument, instead of stuffing it into `AttachOptions`. Thinking here that `containerID` (/name) is a required argument, whereas (most? all?) options should ideally be optional.